src/main/java/org/stellar/sdk/CreatePassiveOfferOperation.java

```diff
@@ -1,135 +1,137 @@
+    private CreatePassiveOfferOperation(Asset selling, Asset buying, String amount, String price) {
+        this.selling = checkNotNull(selling, "selling cannot be null");
+        this.buying = checkNotNull(buying, "buying cannot be null");
+        this.amount = checkNotNull(amount, "amount cannot be null");
+        this.price = checkNotNull(price, "price cannot be null");
+    }
 
     /**
-     * Construct a new CreatePassiveOffer builder from a CreatePassiveOfferOp XDR.
-     * @param op
+     * The asset being sold in this operation
      */
-    Builder(CreatePassiveOfferOp op) {
-      selling = Asset.fromXdr(op.getSelling());
-      buying = Asset.fromXdr(op.getBuying());
-      amount = Operation.fromXdrAmount(op.getAmount().getInt64().longValue());
-      price = Price.fromXdr(op.getPrice()).toString();
+    public Asset getSelling() {
+        return selling;
     }
 
     /**
-     * Creates a new CreatePassiveOffer builder.
-     * @param selling The asset being sold in this operation
-     * @param buying The asset being bought in this operation
-     * @param amount Amount of selling being sold.
-     * @param price Price of 1 unit of selling in terms of buying.
-     * @throws ArithmeticException when amount has more than 7 decimal places.
+     * The asset being bought in this operation
      */
-    public Builder(Asset selling, Asset buying, String amount, String price) {
-      this.selling = checkNotNull(selling, "selling cannot be null");
-      this.buying = checkNotNull(buying, "buying cannot be null");
-      this.amount = checkNotNull(amount, "amount cannot be null");
-      this.price = checkNotNull(price, "price cannot be null");
+    public Asset getBuying() {
+        return buying;
     }
 
     /**
-     * Sets the source account for this operation.
-     * @param sourceAccount The operation's source account.
-     * @return Builder object so you can chain methods.
+     * Amount of selling being sold.
      */
-    public Builder setSourceAccount(KeyPair sourceAccount) {
-      mSourceAccount = checkNotNull(sourceAccount, "sourceAccount cannot be null");
-      return this;
+    public String getAmount() {
+        return amount;
+    }
+
+    /**
+     * Price of 1 unit of selling in terms of buying.
+     */
+    public String getPrice() {
+        return price;
+    }
+
+    @Override
+    org.stellar.sdk.xdr.Operation.OperationBody toOperationBody() {
+        CreatePassiveSellOfferOp op = new CreatePassiveSellOfferOp();
+        op.setSelling(selling.toXdr());
+        op.setBuying(buying.toXdr());
+        Int64 amount = new Int64();
+        amount.setInt64(Operation.toXdrAmount(this.amount));
+        op.setAmount(amount);
+        Price price = Price.fromString(this.price);
+        op.setPrice(price.toXdr());
+
+        org.stellar.sdk.xdr.Operation.OperationBody body = new org.stellar.sdk.xdr.Operation.OperationBody();
+        body.setDiscriminant(OperationType.CREATE_PASSIVE_SELL_OFFER);
```

Comment:
OperationType. CREATE_PASSIVE_OFFER no longer exists, and the actual return here is OperationType. CREATE_PASSIVE_SELL_OFFER.

Reply:
Yes, this is correct. The binary representation is backward compatible.

Reply:
Hi, bartekn, thank you for your guidance.